src/main/java/dk/bec/polonez/reservationsystem/service/OfferService.java

```diff
@@ -2,55 +2,84 @@
 
 
 import dk.bec.polonez.reservationsystem.dto.offerDto.CreateOfferDto;
-import dk.bec.polonez.reservationsystem.dto.offerDto.ResponseOfferDto;
+import dk.bec.polonez.reservationsystem.dto.offerDto.OfferDto;
+import dk.bec.polonez.reservationsystem.dto.offerDto.ResponseOfferFeatureDto;
+import dk.bec.polonez.reservationsystem.model.Feature;
 import dk.bec.polonez.reservationsystem.model.Offer;
-import dk.bec.polonez.reservationsystem.model.User;
+import dk.bec.polonez.reservationsystem.repository.FeatureRepository;
 import dk.bec.polonez.reservationsystem.repository.OfferRepository;
+import dk.bec.polonez.reservationsystem.repository.UserRepository;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
-import org.springframework.web.server.ResponseStatusException;
 
 import java.util.List;
-import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class OfferService {
 
     private final OfferRepository offerRepository;
+    private final FeatureRepository featureRepository;
+    private final ModelMapper modelMapper;
 
     @Autowired
-    public OfferService(OfferRepository offerRepository) {
+    public OfferService(OfferRepository offerRepository, FeatureRepository featureRepository) {
         this.offerRepository = offerRepository;
+        this.featureRepository = featureRepository;
+        this.modelMapper = new ModelMapper();
```

Comment:
**like** I personally like this approach that some beans (the ones that are owned by you and that you will want to mock in unit tests) are injected and some (the types from external lib that you don't own and you dont't want to mock) are initialized in the constructor. By doing it this way your unit test will be easier 'cause you won't have to mock ```ModelWrapper``` type.